torchao/quantization/pt2e/observer.py

```diff
@@ -1891,6 +1891,10 @@ def convert(self, model: torch.fx.GraphModule, observer_node: Node):
             assert self.original_dtype is not None, (
                 "Expecting original_dtype to be populated"
             )
+            # Since input shape & rank may change (e.g. Resnet18), here we need to update block_size for each input
+            self.block_size = get_block_size(
```

Comment:
when does this happen? can you give an example?

Reply:
To reproduce the issue, you may run the code here: https://github.com/pytorch/ao/issues/2094#issue-3010187171
You will have to using `-1` for `block_size` without updating of `self.block_size` here.

Reply:
are you saying the rank / number of dimension changes for input as well? can we use a single `-1` to represent this case?

Reply:
> are you saying the rank / number of dimension changes for input as well?

Yes

> can we use a single -1 to represent this case?

I think it's doable. But there are checks to guard `len(self.block_size) == len(input.shape)`. We need to handle the special case for per-tensor quant at these locations. Is it ok?